src/rebar_prv_install_deps.erl

```diff
@@ -371,31 +374,37 @@ handle_dep(State, DepsDir, AppInfo, Locks, Level) ->
     {AppInfo2, SrcDeps, PkgDeps, Locks++NewLocks, State1}.
 
 -spec maybe_fetch(rebar_app_info:t(), boolean() | {true, binary(), integer()},
-                  sets:set(binary()), rebar_state:t()) -> boolean().
+                  sets:set(binary()), rebar_state:t()) -> {boolean(), rebar_app_info:t()}.
 maybe_fetch(AppInfo, Upgrade, Seen, State) ->
     AppDir = ec_cnv:to_list(rebar_app_info:dir(AppInfo)),
     %% Don't fetch dep if it exists in the _checkouts dir
     case rebar_app_info:is_checkout(AppInfo) of
         true ->
-            false;
+            {false, AppInfo};
         false ->
             case rebar_app_discover:find_app(AppDir, all) of
                 false ->
                     case in_default(AppInfo, State) of
                         false ->
-                            fetch_app(AppInfo, AppDir, State);
+                            {fetch_app(AppInfo, AppDir, State), AppInfo};
                         {true, FoundApp} ->
+                            %% Preserve the state we created with overrides
+                            AppState = rebar_app_info:state(AppInfo),
+                            FoundApp = rebar_app_info:state(FoundApp, AppState),
```

Comment:
Is this even gonna work? Is this untested or an assertion? If it's an assertion, why reassign it after?